mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -327,7 +329,17 @@ inline void emitStoreLogic(OpBuilder &b, Location loc, MemRefType destType,
                                             ifWithinBoundsOp.getResults()[4],
                                             ifWithinBoundsOp.getResults()[5]});
   } else {
-    b.create<StoreOp>(loc, value, dest, destLowerIndices);
+    if (memoryOp == InMemoryDataOperation::Set)
+      b.create<StoreOp>(loc, value, dest, destLowerIndices);
+    else {
+      SmallVector<Value, 8> destLowerStoreIndices;
+      for (unsigned i = 0; i < destLowerIndices.size(); ++i) {
+        auto dstIndex = b.create<IndexCastOp>(loc, destLowerIndices[i],
+                                              b.getIntegerType(32));
+        destLowerStoreIndices.push_back(dstIndex);
+      }
+      b.create<gpu::AtomicFAddOp>(loc, value, dest, destLowerStoreIndices);
```

Comment:
`atomicFAdd` is only used in non-OOB cases in the current implementation, would it be used in OOB cases?

Reply:
for backward weight, OOB is not required